unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/Font/Providers/AddressableFontProvider.cs

```diff
@@ -0,0 +1,11 @@
+using Cysharp.Threading.Tasks;
+using DCL.Providers;
+using System;
+using System.Threading;
+using TMPro;
+
+public class AddressableFontProvider : AddressableResourceProvider<TMP_FontAsset>, IFontAssetProvider
```

Comment:
Consider using `Composition` over `Inheritance` to enrich API. I see several reasons for that:
1. `AddressableResourceProvider` is a singleton (service) by nature
2. In your case `IAddressableResourceProvider<T>` is redundant, e.g. if you were introducing tests for this piece you could not mock the underlying API
3. `T` argument looks redundant in `AddressableResourceProvider<T>` and `IAddressableResourceProvider<T>` (you don't have state there!), you can make methods generic instead and reuse that `Service` for every application

Reply:
Done, but I would like your opinion on it

Reply:
This is exactly what I meant (apart of the interface usage)